### DISCUSSION TITLE
easiest/easier way to find dates covered by a given .bsp?

### DISCUSSION THREAD

**Discussion by davidmikolas** [skyfielders/python-skyfield#566] [Ideas]
I'm not sure if this belongs here as a discussion or should be posted as an issue, I'll try it here first.

I've just loaded de405 because Neptune can't even orbit the Sun once in de421. Now I want to know the first/last dates covered by de405 in some nicer way than trying some dates and getting errors.

    eph = loaddata('de405.bsp')
    comments = eph.comments()

returns a ~4,400 character string and I can print it and scroll through and find

        BEGIN_TIME      = CAL-ET 1599 DEC 09 00:00:00.000
        END_TIME        = CAL-ET 2201 FEB 20 00:00:00.000

but is there a more direct and/or easy way to get these times using python?  Maybe `ts(eph.begin_time)` and `ts(eph.end_time)`?

**Comment by brandon-rhodes** [comment]
Take a look at the `segments` list — there's a bit of sample code at the bottom of:

https://rhodesmill.org/skyfield/planets.html#choosing-an-ephemeris

I think you'll find that `segment.time_range(ts)` returns the pair of dates; try that out and see if they provide you with the data you need.

**Reply by davidmikolas** [reply]
Thanks!

For some reason I can never wrap my head around time objects.

Okay I have:

    from skyfield.api import Loader
    loaddata = Loader('~/Documents/SkyData')
    ts = loaddata.timescale()   # include builtin=True if you want to use older files (you may miss some leap-seconds)
    eph = loaddata('de405.bsp')   # need bigger one because Neptune too slow for de421.bsp
    begin, end = eph.segments[0].time_range(ts)[0], eph.segments[-1].time_range(ts)[1]
    print(begin, end)

There is 

    ts.J(np.linspace(begin.utc.year+1, end.utc.year-1, 10000))   # +1, -1 for safety

which seems to be good enough for me (i.e. problem solved) but I wonder if there is something more elegant?
